go/mysql/mysql56_gtid_set.go

```diff
@@ -435,6 +435,128 @@ func (set Mysql56GTIDSet) SIDBlock() []byte {
 	return buf.Bytes()
 }
 
+func (set Mysql56GTIDSet) Difference(other Mysql56GTIDSet) Mysql56GTIDSet {
+	if other == nil {
```

Comment:
Does it make sense to check if `set` is nil here? It's perfectly valid in go, although often confusing.

Reply:
I think it is valid because `Mysql56GTIDSet` being a map is a pointer type. If it's nil then the difference is the receiver.

Reply:
Oh wait, I see what you're saying! Sorry, I got confused. You meant in addition. I think that makes sense. If set is nil, then we should return nil. I'll add the check

Reply:
Done.